node/src/actors/session/handlers.rs

```diff
@@ -116,14 +116,35 @@ impl StreamHandler<BytesMut, Error> for Session {
                     (
                         _,
                         SessionStatus::Unconsolidated,
-                        Command::Version(Version { sender_address, .. }),
+                        Command::Version(Version {
+                            sender_address,
+                            timestamp,
+                            ..
+                        }),
                     ) => {
-                        let msgs = handshake_version(self, &sender_address);
-                        for msg in msgs {
-                            self.send_message(msg);
+                        let current_ts = get_timestamp();
+                        let received_ts = timestamp;
+
+                        if self.handshake_max_ts_diff != 0
+                            && (current_ts - received_ts).abs() > self.handshake_max_ts_diff
+                        {
+                            log::warn!(
+                                "Invalid timestamp in Handshake Version message received.\n\
```

Comment:
It would be nice to see the difference in the log:

```suggestion
                                "Dropping peer because their timestamp is too far away (±{} seconds)
```

"too far away" may not be the best words, maybe "because the difference between timestamps is too big" or something.